### PR TITLE
#3001 Display an info log if z is NaN

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [x] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Modeler/Common/DYNModelMulti.cpp [hunk 1]
```diff
@@ -493,9 +493,13 @@ ModelMulti::propagateZModif() {
...
-    if (doubleNotEquals(zLocal_[nonSilentZIndexes_[i]], zSave_[nonSilentZIndexes_[i]])) {
-      indicesDiff.push_back(static_cast<int>(nonSilentZIndexes_[i]));
-      zChangeType = NOT_SILENT_Z_CHANGE;
+    if (!std::isnan(zLocal_[nonSilentZIndexes_[i]]) && !std::isnan(zSave_[nonSilentZIndexes_[i]])) {
+      if (doubleNotEquals(zLocal_[nonSilentZIndexes_[i]], zSave_[nonSilentZIndexes_[i]])) {
+        indicesDiff.push_back(static_cast<int>(nonSilentZIndexes_[i]));
+        zChangeType = NOT_SILENT_Z_CHANGE;
+      }
+    } else {
+      Trace::info() << DYNLog(ZValueIsNaN) << Trace::endline;
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/Common/DYNModelMulti.cpp:502]: We want to crash at this point as if a Z is a NaN then something is wrong

File: dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic [hunk 2]
```diff
@@ -156,6 +156,7 @@ DynamicLineStatusNotSupported =           connection status (CLOSED_1 or CLOSED_
 MismatchingVariableSizes    =             the number of %1% variables and corresponding size are different in model %2% (%3% != %4%)
 ConverterWrongType          =             wrong converter type for converter %1%
 AttemptToPropagateBeforeMerge        =    discrete variables connectors should be merged before propagating information to model
+ZValueIsNaN                 =             z value is NaN
```
Comment (rosiereflo) [inline dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic:159]: Please provide more information (at least the model name and the variable index)

File: dynawo/sources/Modeler/Common/DYNModelMulti.cpp [hunk 3]
```diff
@@ -493,9 +493,13 @@ ModelMulti::propagateZModif() {
   vector<int> indicesDiff;
   zChangeType_t zChangeType = NO_Z_CHANGE;
   for (std::size_t i = 0, iEnd = nonSilentZIndexes_.size(); i < iEnd; ++i) {
-    if (doubleNotEquals(zLocal_[nonSilentZIndexes_[i]], zSave_[nonSilentZIndexes_[i]])) {
-      indicesDiff.push_back(static_cast<int>(nonSilentZIndexes_[i]));
-      zChangeType = NOT_SILENT_Z_CHANGE;
+    if (!std::isnan(zLocal_[nonSilentZIndexes_[i]]) && !std::isnan(zSave_[nonSilentZIndexes_[i]])) {
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/Common/DYNModelMulti.cpp:496]: The test should be done in the loop after also

File: dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic (hunk 2 above)
Comment (barondim) [reply dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic:159]: This error is thrown in ModelMulti::propagateZModif() and is not associated to any model

File: dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic [hunk 4]
```diff
@@ -156,6 +156,7 @@ DynamicLineStatusNotSupported =           connection status (CLOSED_1 or CLOSED_
 MismatchingVariableSizes    =             the number of %1% variables and corresponding size are different in model %2% (%3% != %4%)
 ConverterWrongType          =             wrong converter type for converter %1%
 AttemptToPropagateBeforeMerge        =    discrete variables connectors should be merged before propagating information to model
+ZValueIsNaN                 =             z value is NaN in discrete variables buffer at index %1%
```
Comment (rosiereflo) [inline dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic:159]: I would propose "discrete variable with index %1% has a NaN value"

Comment (barondim) [reply dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic]: I agree, it's more neat

**Review by rosiereflo — APPROVED** [review]

**Comment by barondim** [comment]
- [ ] NRT don't pass anymore. Do not merge.

**Pull request merged**